### PR TITLE
Correct behaviour of sort in `concat` for singleton concatenations

### LINKED ISSUE #12132: [BUG] `cudf.concat` with `sort=True` does not match pandas behaviour when concatenating a single object
**Describe the bug**

The pandas documentation for [`pandas.concat`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html) says, of `sort`:

> Sort non-concatenation axis if it is not already aligned when join is ‘outer’. This has no effect when join='inner', which already preserves the order of the non-concatenation axis.

So if we pass `axis=0` then we expect axis-1 to be sorted (and vice versa).

Aside: `sort=True` applies in all cases (irrespective of whether `join="inner"` or `"outer"`) in pandas as far as I can tell (filed https://github.com/pandas-dev/pandas/issues/49646 to check)

However, if we are only concatenating a singleton list of dataframes, this logic is backwards

https://github.com/rapidsai/cudf/blob/d335aa3dcf6a3c55dc7dbd95b94ca8e4b409daee/python/cudf/cudf/core/reshape.py#L278

We want to sort on `1 - axis`.

**Steps/Code to reproduce bug**

```python
import cudf
df = cudf.DataFrame({"b": [1, 2], "a": [3, 4]}, index=["C", "A"])

df2 = cudf.concat([df], sort=True, join="inner")
print(df2)
#    b  a
# A  2  4
# C  1  3

# Expected output
#    a  b
# C  3  1
# A  4  2
```

**Expected behavior**

concat should sort the appropriate axis.

### PR DESCRIPTION
**Pull request by wence-**
## Description

The sort argument to `concat` requests a sort of the non-concatenated axis.
Previously this was handled correctly except when passing only a single object
to concatenate.

Closes #12132.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_concat.py [hunk 1]
```diff
@@ -1793,3 +1793,48 @@ def test_concat_categorical_ordering():
...
+            {
+                "b": [1, 2, 3, 4],
+                "d": [7, 8, 9, 10],
+                "a": [4, 5, 6, 7],
+                "c": [10, 11, 12, 13],
+            },
+            index=concat_index,
+        )
+    else:
+        return pd.Series([4, 5, 5, 6], index=concat_index)
```
Comment (shwina) [inline python/cudf/cudf/tests/test_concat.py:1819]: This seems like a reasonable use of fixtures to me, except `concat_index` and `concat_obj` seem like very general names for fixtures that are only ever used by a single test. If the intent is to use them more generally, should they appear near the top of this file?

File: python/cudf/cudf/tests/test_concat.py [hunk 2]
```diff
@@ -1793,3 +1793,48 @@ def test_concat_categorical_ordering():
...
+def concat_index(request):
+    if request.param == "rangeindex":
+        return pd.RangeIndex(0, 4)
+    else:
+        return pd.Index(["a", "h", "g", "f"])
+
+
+@pytest.fixture(params=["dataframe", "series"])
+def concat_obj(request, concat_index):
+    if request == "dataframe":
```
Comment (shwina) [inline python/cudf/cudf/tests/test_concat.py:1808]:
```suggestion
    if request.param == "dataframe":
```

File: python/cudf/cudf/tests/test_concat.py (hunk 1 above)
Comment (wence-) [reply python/cudf/cudf/tests/test_concat.py:1819]: I'm not sure they're more generic (renamed to `singleton_XXX`)

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**